cartoframes/dataset.py

```diff
@@ -55,32 +58,37 @@ def __init__(self, table_name=None, schema='public', query=None, df=None, gdf=No
             self.normalized_column_names = _normalize_column_names(gdf)
 
     @classmethod
-    def from_table(cls, table_name, schema='public', context=None):
-        return cls(
-            table_name=table_name, schema=schema, context=context, state=cls.STATE_REMOTE)
+    def from_table(cls, table_name, context, schema='public'):
+        return cls(table_name=table_name, schema=schema, context=context, state=cls.STATE_REMOTE)
 
     @classmethod
-    def from_query(cls, query, context=None):
-        return cls(
-            query=query, context=context, state=cls.STATE_REMOTE)
+    def from_query(cls, query, context):
+        return cls(query=query, context=context, state=cls.STATE_REMOTE)
 
     @classmethod
-    def from_dataframe(cls, df, table_name=None, schema='public', context=None):
-        return cls(
-            df=df, table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_dataframe(cls, df):
+        return cls(df=df, state=cls.STATE_LOCAL)
 
     @classmethod
-    def from_geodataframe(cls, gdf, table_name=None, schema='public', context=None):
-        return cls(
-            gdf=gdf, table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_geodataframe(cls, gdf):
+        return cls(gdf=gdf, state=cls.STATE_LOCAL)
 
     @classmethod
-    def from_geojson(cls, geojson, table_name=None, schema='public', context=None):
-        return cls(
-            gdf=load_geojson(geojson), table_name=table_name, schema=schema, context=context, state=cls.STATE_LOCAL)
+    def from_geojson(cls, geojson):
+        return cls(gdf=load_geojson(geojson), state=cls.STATE_LOCAL)
+
+    def upload(self, with_lonlat=None, if_exists='fail', table_name=None, schema=None, context=None):
+        if table_name:
+            self.table_name = table_name
```

Comment:
I guess we should normalize the `table_name`
```suggestion
            self.table_name = normalize_name(table_name)
```

Reply:
It is true!